app/routes/routes.py

```diff
@@ -46,193 +15,44 @@ def lore():
-        gender = site_session.get("gender")
-        profile_description = site_session.get("profile_description")
-        domicile = site_session.get("domicile")
-        education = site_session.get("education")
-        employment_status = site_session.get("employment_status")
-        type_of_robot = site_session.get("type_of_robot")
-        distance = site_session.get("distance")
-        employment_status_criteria = site_session.get("employment_status_criteria")
-
-        image_name = get_image(form.photo.data, name)
-
-        insert_user_and_user_profile(
-            username,
-            name,
-            lastname,
-            password,
-            email,
-            image_name,
-            age,
-            gender,
-            profile_description,
-            domicile,
-            education,
-            employment_status,
-            type_of_robot,
-            distance,
-            employment_status_criteria,
-        )
-
-        user = get_user_from_User_table(username)
-
-        login_user(user)
-
-        session.pop("username", None)
-        session.pop("name", None)
-        session.pop("lastname", None)
-        session.pop("email", None)
-        session.pop("password", None)
-        session.pop("age", None)
-        session.pop("gender", None)
-        session.pop("profile_description", None)
-        session.pop("domicile", None)
-        session.pop("education", None)
-        session.pop("employment_status", None)
-        session.pop("type_of_robot", None)
-        session.pop("distance", None)
-        session.pop("employment_status_criteria", None)
-
-        return redirect(url_for("user_homepage"))
-    return render_template("get-main-image.html", form=form)
-
 """
 @app.route("/user_homepage", methods=["GET", "POST"])
 def user_homepage():
     logout_user()
     return redirect(url_for("home"))
 """
+
+
 @app.route("/user_homepage", methods=["GET"])
 @login_required
 def user_homepage():
     print(current_user.username)
-    if current_user.is_authenticated:
-        return render_template("user_homepage.html", user=current_user.name)
-    
+    return render_template("user_homepage.html", user=current_user.name)
+
+
 @app.route("/get_geolocation", methods=["POST"])
 @login_required
 def get_geolocation():
     data = request.get_json()
     print(current_user.username)
-    if data:
-        latitude = data.get('latitude')
-        longitude = data.get('longitude')
-        if latitude and longitude:
-            print(f"{latitude}, {longitude}")
-            update_user_location(longitude, latitude, current_user.username)
-            user = get_user_from_User_table(current_user.username)
-            print(user)
-            return redirect(url_for("user_homepage"))
+
+    latitude = data.get("latitude")
+    longitude = data.get("longitude")
+    print(f"{latitude}, {longitude}")
+
+    update_user_location(longitude, latitude, current_user)
+    user = get_user(current_user.username)

```

Comment:
isnt it the same user? why update_user_location can't return updated user?

Reply:
If i am correct this is updated user.